test/wasm2js/float_literals-modified.2asm.js

```diff
@@ -42,123 +42,153 @@ function asmFunc(global, env, buffer) {
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $2() {
-  return (wasm2js_scratch_store_f32(Math_fround(-nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(-nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $3() {
-  return (wasm2js_scratch_store_f32(Math_fround(nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $4() {
-  return (wasm2js_scratch_store_f32(Math_fround(nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $5() {
-  return (wasm2js_scratch_store_f32(Math_fround(-nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(-nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $6() {
-  return (wasm2js_scratch_store_f32(Math_fround(nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $7() {
-  return (wasm2js_scratch_store_f32(Math_fround(nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $8() {
-  return (wasm2js_scratch_store_f32(Math_fround(-nan)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(-nan)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $9() {
-  return (wasm2js_scratch_store_f32(Math_fround(infinity)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(infinity)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $10() {
-  return (wasm2js_scratch_store_f32(Math_fround(infinity)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(infinity)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $11() {
-  return (wasm2js_scratch_store_f32(Math_fround(-infinity)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(-infinity)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $12() {
-  return (wasm2js_scratch_store_f32(Math_fround(0.0)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(0.0)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $13() {
-  return (wasm2js_scratch_store_f32(Math_fround(0.0)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(0.0)));
+  return wasm2js_scratch_load_i32(0 | 0) | 0 | 0;
  }
  
  function $14() {
-  return (wasm2js_scratch_store_f32(Math_fround(-0.0)), wasm2js_scratch_load_i32(0)) | 0;
+  wasm2js_scratch_store_f32(Math_fround(Math_fround(-0.0)));
```

Comment:
I see everywhere `Math_fround(Math_fround(...))`. Isn't this redundant?

Reply:
Yes, that is redundant. The optimized output should not have that, though - if it does, that's a bug in the optimizer in `wasm2js.cpp`.